Upstream visionOS media controls

hortont424 opened the pull request:
#### 147b9c2e24ab34b75aa86fa8b0bb2e8f6ca4bad8
<pre>
Upstream visionOS media controls
<a href="https://bugs.webkit.org/show_bug.cgi?id=257920">https://bugs.webkit.org/show_bug.cgi?id=257920</a>
rdar://110552296

Reviewed by Megan Gardner.

* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp:
(WebCore::MediaControlsHost::layoutTraitsClassName const):
* Source/WebCore/Modules/modern-media-controls/controls/vision-layout-traits.js: Added.
* Source/WebCore/Modules/modern-media-controls/controls/vision-media-controls.css: Added.
* Source/WebCore/Modules/modern-media-controls/controls/vision-media-controls.js: Added.
* Source/WebCore/Modules/modern-media-controls/controls/vision-slider.css: Added.
* Source/WebCore/Modules/modern-media-controls/controls/vision-slider.js: Added.

Canonical link: <a href="https://commits.webkit.org/265043@main">https://commits.webkit.org/265043@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ef18033971785cc39bcbe06766d1c5cac2fa6d70

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [  ~~🧪 style~~](https://ews-build.webkit.org/#/builders/38/builds/9614 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/26/builds/9870 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/14/builds/10114 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/11275 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/9399 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [  ~~🧪 bindings~~](https://ews-build.webkit.org/#/builders/9/builds/9624 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/23/builds/11849 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/16/builds/9825 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/11275 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/11/builds/9766 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/23/builds/11849 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/14/builds/10114 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/11434 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/23/builds/11849 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/14/builds/10114 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/11434 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/23/builds/11849 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/14/builds/10114 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/11434 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/9378 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/16/builds/9825 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/30/builds/8569 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/14/builds/10114 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/12790 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1103 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/31/builds/9152 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebCore/DerivedSources-input.xcfilelist:
```diff
@@ -1493,7 +1498,6 @@ $(PROJECT_DIR)/mathml/mathtags.in
 $(PROJECT_DIR)/page/BarProp.idl
 $(PROJECT_DIR)/page/Crypto.idl
 $(PROJECT_DIR)/page/DOMSelection.idl
-$(PROJECT_DIR)/page/DOMWindow.idl
```
This is a duplicate and the script removed it...

megangardner commented on Source/WebCore/DerivedSources.make:
```diff
@@ -1884,9 +1875,10 @@ MODERN_MEDIA_CONTROLS_STYLE_SHEETS = \
     $(WebCore)/Modules/modern-media-controls/controls/text-tracks.css \
     $(WebCore)/Modules/modern-media-controls/controls/time-label.css \
+	$(WebCore)/Modules/modern-media-controls/controls/vision-media-controls.css \
+	$(WebCore)/Modules/modern-media-controls/controls/vision-slider.css \
```
Are these double-indented on purpose?

hortont424 replied:
I think github is lying to you

nt1m replied:
it's not, this looks tab-indented, the other lines are space indented

hortont424 replied:
INTERESTING. I guess my github is configured to use 4 space tabs? (and my editor certainly is). I thought Makefiles were always tab-indented, though...

megangardner commented on Source/WebCore/DerivedSources.make:
```diff
@@ -1992,6 +1975,9 @@ MODERN_MEDIA_CONTROLS_SCRIPTS = \
     $(WebCore)/Modules/modern-media-controls/controls/pip-placard.js \
+	$(WebCore)/Modules/modern-media-controls/controls/vision-slider.js \
+	$(WebCore)/Modules/modern-media-controls/controls/vision-media-controls.js \
+	$(WebCore)/Modules/modern-media-controls/controls/vision-layout-traits.js \
```
same as above, or is GitHub showing me things wrong?

nt1m replied:
ditto

hortont424 committed:
Upstream visionOS media controls
https://bugs.webkit.org/show_bug.cgi?id=257920
rdar://110552296

Reviewed by Megan Gardner.

* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp:
(WebCore::MediaControlsHost::layoutTraitsClassName const):
* Source/WebCore/Modules/modern-media-controls/controls/vision-layout-traits.js: Added.
* Source/WebCore/Modules/modern-media-controls/controls/vision-media-controls.css: Added.
* Source/WebCore/Modules/modern-media-controls/controls/vision-media-controls.js: Added.
* Source/WebCore/Modules/modern-media-controls/controls/vision-slider.css: Added.
* Source/WebCore/Modules/modern-media-controls/controls/vision-slider.js: Added.

Canonical link: https://commits.webkit.org/265043@main
